From d72ef9a075dc7116feff256d941a64aa49fb9350 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:56 -0400
Subject: [usb] add a private-data pointer to struct usb_tt

Message-id: <1410887463-20674-45-git-send-email-dzickus@redhat.com>
Patchwork-id: 95161
O-Subject: [RHEL7 PATCH 044/591] USB: add a private-data pointer to struct usb_tt
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 7c4bb942986fc2aa7ca4fccfed665d24525a0e21
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Oct 11 11:29:22 2013 -0400

    USB: add a private-data pointer to struct usb_tt

    For improved scheduling of transfers through a Transaction Translator,
    ehci-hcd will need to store a bunch of information associated with the
    FS/LS bus on the downstream side of the TT.  This patch adds a pointer
    for such HCD-private data to the usb_tt structure.

    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/usb/hcd.h b/include/linux/usb/hcd.h
index dbe3cd1..b8aba19 100644
--- a/include/linux/usb/hcd.h
+++ b/include/linux/usb/hcd.h
@@ -506,6 +506,7 @@ struct usb_tt {
  struct usb_device *hub; /* upstream highspeed hub */
  int   multi; /* true means one TT per port */
  unsigned  think_time; /* think time in ns */
+ void   *hcpriv; /* HCD private data */
 
  /* for control/bulk error recovery (CLEAR_TT_BUFFER) */
  spinlock_t  lock;
-- 
1.7.1